### DISCUSSION TITLE
Adding support for XGBoost

### DISCUSSION THREAD

**Discussion by tunguz** [pyodide/pyodide#2486] [Ideas]
I was wondering if it would be possible to add support for XGBoost in pyodide? In my experience XGBoost is generally very easy to install - I've installed it on almost every system that I've used, from small Raspberry Pi Zeros, to large DGX Stations.

**Comment by hoodmane** [comment]
It seems we have to update cmake first. @henryiii was saying we should just `pip install cmake` instead of installing the ancient version from apt.

**Reply by tunguz** [reply]
That would be great. How hard would that be to pull off?

**Reply by tunguz** [reply]
Right now I just need a simple demo of XGBoost working with PyScript.

**Reply by hoodmane** [reply]
I think not hard. Currently we install cmake here:
https://github.com/pyodide/pyodide/blob/f04ba9d56a1e7be13e8cb796b6fe9252c0dd25b5/Dockerfile#L7
So we need to remove that and add it to `requirements.txt` and then rebuild the docker image. If you want to move this along, you could make a PR doing this. You can deploy the docker image to your personal dockerhub account and update the image here:
https://github.com/pyodide/pyodide/blob/f04ba9d56a1e7be13e8cb796b6fe9252c0dd25b5/.circleci/config.yml#L9
and here:
https://github.com/pyodide/pyodide/blob/f04ba9d56a1e7be13e8cb796b6fe9252c0dd25b5/run_docker#L3

If CI passes, then we will push the updated image to pyodide's docker hub, update the images again, and merge.

**Reply by henryiii** [reply]
I've love the new CMake too. :) Why is clang-format there? Is that historical?

**Reply by hoodmane** [reply]
I guess we're now using clang-format from precommit? If so then yeah it's historical. Removing that should have a positive impact on our image size since I think it pulls in `clang-6.0` and other stuff too.

**Reply by tunguz** [reply]
Sorry, I am actually not that familiar with Docker. Would need to ask around for some help, unless you guys have some bandwidth to go ahead and implement this.

**Reply by henryiii** [reply]
Shouldn't there be a `&& rm -rf /var/lib/apt/lists/*` at the end of that line?

**Reply by henryiii** [reply]
https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#apt-get

**Reply by henryiii** [reply]
Yes, pre-commit is getting clang-format from a <2MB manylinux wheel.

**Reply by hoodmane** [reply]
> `&& rm -rf /var/lib/apt/lists/*`

Uh... yes. I can probably go ahead and do this, it's a simple enough task.

**Reply by tunguz** [reply]
Hi guys, happy Friday. Just checking to see if there has been any progress with updating make. Thanks. 😄

**Reply by hoodmane** [reply]
Yeah @henryiii did it #2489

**Reply by tunguz** [reply]
Woohoo, great! Can we have XGBoost now? Please!!! 😀

**Reply by hoodmane** [reply]
Well it seems like there are still some remaining issues around how to handle CMake toolchains, cf the related discussion here:
https://github.com/pyodide/pyodide/pull/2492
once we figure out the right way to handle these various CMake issues, hopefully we will be able to build xgboost and other CMake-based packages.

**Reply by tunguz** [reply]
Great, thanks for letting me know!

**Reply by tunguz** [reply]
Hi @hoodmane and @henryiii , just wanted to see if you have been able to resolve the issues with CMake toolchains that were blocking you from supporting XGBoost in pyodide?

**Reply by henryiii** [reply]
Has it been retried? The current issue I'm working on is specific to pybind11's classic FindPythonInterp/Libs support, and I think the check can be disabled from pybind11. XGBoost might work now?

**Reply by tunguz** [reply]
Do you have any bandwidth to retry installing it?

**Reply by henryiii** [reply]
Looks like it was retried (see https://github.com/pyodide/pyodide/pull/2537) and ran into the problem that it doesn't seem to support running in a single thread.

**Reply by tunguz** [reply]
Oh I see. That's a bummer. 😢

**Reply by tunguz** [reply]
@henryiii I've asked around, and there seems to be a way to install a single-threaded version of XGBoost. Apparently you just need to set the CMake flag `USE_OPENMP=OFF`. Works with and without GPU.

**Reply by henryiii** [reply]
Nice, @ryanking13 did you try that?
